frontend/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx

```diff
@@ -468,14 +468,18 @@ function dataIsAnAppendOfPrev(
 ): boolean {
   // Check whether dataframes have the same shape.
 
+  // not an append
   if (prevNumCols !== numCols) {
     return false
   }
 
+  // an append would have prev number of rows < new number of rows
+  // the = handles cases of dynamic inputs to data having same number of rows as previous
```

Comment:
This is a little confusing. How about "Data can be updated, but still have the same number of rows. Thus, we consider the case an append only when the number of rows has increased."

Reply:
Agreed, that sounds a lot better, thanks!